etl/steps/data/garden/un/2021-12-20/un_igme.py

```diff
@@ -0,0 +1,240 @@
+import json
+from typing import List, cast
+
+import pandas as pd
+from owid import catalog
+from owid.catalog import Dataset, Source, Table
+from owid.catalog.utils import underscore_table
+from owid.datautils import geo
+from structlog import get_logger
+
+from etl.helpers import Names
+from etl.paths import DATA_DIR
+
+log = get_logger()
+
+GAPMINDER_CHILD_MORTALITY_DATASET_PATH = DATA_DIR / "open_numbers/open_numbers/latest/gapminder__child_mortality"
+GAPMINDER_INFANT_MORTALITY_DATASET_PATH = DATA_DIR / "open_numbers/open_numbers/latest/gapminder__hist_imr"
+
+
+# naming conventions
+N = Names(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("un_igme.start")
+    # read dataset from meadow
+    ds_meadow = Dataset(DATA_DIR / "meadow/un/2021-12-20/un_igme")
+    tb_meadow = ds_meadow["un_igme"]
+    df = pd.DataFrame(tb_meadow).drop(columns=["index"])
+    # adding source tag to all UN IGME rows prior to combination with Gapminder data
+    df["source"] = "UN IGME"
+    df_gap = get_gapminder_data(max_year=max(df["year"]))
+    df_combine = pd.concat([df, df_gap])
+
+    log.info("un_igme.exclude_countries")
+    df_combine = exclude_countries(df_combine)
+
+    log.info("un_igme.harmonize_countries")
+    dfc = harmonize_countries(df_combine)
+    # Preferentially use UN IGME data where there is duplicate values for country-year combinations
+    dfc = combine_datasets(dfc)
+    # Calculate missing age-group mortality rates
+    dfc = calculate_mortality_rate(dfc)
+
+    # Calculate the absolute number of youth deaths
+    dfc = calculate_youth_deaths(dfc)
+
+    # Making the values in the table a bit more appropriate for our use and pivoting to a wide table.
+    log.info("un_igme.clean_data")
+    dfc = clean_and_format_data(dfc)
+
+    ds_garden = Dataset.create_empty(dest_dir)
+    ds_garden.metadata = ds_meadow.metadata
+
+    tb_garden = Table(dfc)
+    tb_garden.metadata = tb_meadow.metadata
+    # Flatten the indexed columns
+    tb_garden.columns = ["__".join(col).strip() for col in tb_garden.columns.values]
+    # Create one table per column and auto-populate the metadata
+    for col in tb_garden.columns:
```

Comment:
Instead of creating columns yourself, you could move some columns to dimensions and things like title will be generated automatically. Here's a sample code

```
tb_garden = Table(dfc, metadata=tb_meadow.metadata)
tb_garden.columns.names = ["metric", "sex", "indicator", "unit"]

# move sex and metric to dimensions
tb_garden = tb_garden.stack(level="metric").stack(level="sex")

# add units to variable metadata
units = tb_garden.columns.get_level_values("unit")
tb_garden.columns = tb_garden.columns.droplevel(level="unit")

for col, unit in zip(tb_garden.columns, units):
    tb_garden[col].metadata.unit = unit
```

Reply:
Thanks - this is much nicer!